### PR TITLE
31280 Legacy iOS mobile apps crashing

### PR DESCRIPTION
**Pull request by jdrichards50**
Allow date and time parameters to be processed correctly. The date and time fields are just text boxes. We don't have the time to spend on the legacy apps to implement a date and time picker.

### CODE REVIEW

**Review by mwilber-ciq — COMMENTED** [review]
This looks good. I agree with your comment that it's not worth putting the date & time pickers in but the string of numbers in the text field is a little confusing. Is it possible to modify the field label to add "(Format: YYYYMMDD)" and "(Format: HHMMSS)"?

Also, what about the `Charting-Library---iOS-Sample-App` repo? Does that need to be updated as well? The docs are still linking to it.

**Comment by jdrichards50** [comment]
> This looks good. I agree with your comment that it's not worth putting the date & time pickers in but the string of numbers in the text field is a little confusing. Is it possible to modify the field label to add "(Format: YYYYMMDD)" and "(Format: HHMMSS)"?
> 
> Also, what about the `Charting-Library---iOS-Sample-App` repo? Does that need to be updated as well? The docs are still linking to it.

I can probably add something to the label. iOS-Sample-App will need to be updated as well, but it also needs a new cocoa pod, so I'll just update those after this one is approved.

File: ChartIQDemo/ChartIQDemo/StudyDetailViewController.swift [hunk 1]
```diff
@@ -241,9 +245,15 @@ class StudyDetailViewController: UITableViewController {
...
             cell.labels?[1].text = (parameter["value"] as? String)?.capitalized ?? ""
-        case .number, .text:
+        case .number, .text, .date, .time:
             cell.textFields![0].keyboardType = optionType == .number ? .numberPad : .default
             cell.textFields?[0].text = optionType == .number ? String(parameter["value"] as? Float ?? 0) : parameter["value"] as? String ?? ""
+            if(optionType == .date) {
+                cell.labels?[0].text! += " (yyyymmdd)"
+            }
+            if(optionType == .time) {
+                cell.labels?[0].text! += " (hh:mm:ss)"
```
Comment (mwilber-ciq) [inline ChartIQDemo/ChartIQDemo/StudyDetailViewController.swift]: Should the colons be here?

**Review by mwilber-ciq — COMMENTED** [review]
Looks like the label is used to identify the field. Now it won't save changes to the date & time. If this is going to be a can of worms I think it's best not to pursue seeing as how this app will be deprecated soon.

File: ChartIQDemo/ChartIQDemo/StudyDetailViewController.swift (hunk 1 above)
Comment (jdrichards50) [reply ChartIQDemo/ChartIQDemo/StudyDetailViewController.swift]: > Looks like the label is used to identify the field. Now it won't save changes to the date & time. If this is going to be a can of worms I think it's best not to pursue seeing as how this app will be deprecated soon.

Heh, of course it breaks it... Good thing you checked. I say revert and just leave the label as is. It's not worth it to make a check on the sdk side.

Comment (mwilber-ciq) [reply ChartIQDemo/ChartIQDemo/StudyDetailViewController.swift]: How are we handling these fields in the new app? Is there a datetime picker?

**Review by mwilber-ciq — APPROVED** [review]

**Pull request merged**